### PR TITLE
feat(lint/noUselessElse): add rule

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

Implements #247.

I am not 100% sold on the rule name. I think that `noUselessElse` doesn't communicate the actual intent of the rule. Unfortunately I didn't find a better name yet. Clippy uses `no_redundant_else`.

Trivia handling is again a pain. I finally gave up and keep the suggested code mis-indented.
We should certainly implement a kind of "mini formatter" for the code fixes.
An alternative is to remove the code fix of the rule...

## Test Plan

Tests included

### CODE REVIEW

**Comment by ematipico** [comment]
> Unfortunately I didn't find a better name yet. Clippy uses `no_redundant_else`.

I don't mind clippy's name :)

File: crates/biome_js_analyze/src/analyzers/nursery/no_useless_else.rs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                AnyJsStatement::JsIfStatement(if_stmt) => {
+                    let Some(else_clause) = if_stmt.else_clause() else {
+                        // No else clause
+                        return None;
+                    };
+                    stmt_stack.push((if_stmt.consequent().ok()?, metadata));
+                    stmt_stack.push((else_clause.alternate().ok()?, metadata));
+                }
+                AnyJsStatement::JsSwitchStatement(switch_stmt) => {
+                    // To simplify, We do not tale fallthoughs into account.
```
Comment (unvalley) [inline crates/biome_js_analyze/src/analyzers/nursery/no_useless_else.rs:108]: nits:
```suggestion
                    // To simplify, We do not take fallthroughs into account.
```

File: crates/biome_js_analyze/src/analyzers/nursery/no_useless_else.rs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow `else` block when the `if` block breaks early.
+    ///
+    /// If an `if` block breaks early using a breaking statement (`return`, `break`, `continue`, or `throw`),
+    /// then the `else` block becomes useless.
+    /// Its contents can be placed outside of the block.
+    ///
```
Comment (unvalley) [inline crates/biome_js_analyze/src/analyzers/nursery/no_useless_else.rs:18]: should we refer the ESLint [no-else-return](https://eslint.org/docs/latest/rules/no-else-return/)?

Comment (Conaclos) [reply crates/biome_js_analyze/src/analyzers/nursery/no_useless_else.rs:18]: I don't think so. Although the rule covers `no-elese-return`, there is too much difference?

**Comment by Conaclos** [comment]
> maybe a chance to create naming rule for lints

I personally prefer `Useless` to `Redundant`. Not sure if we should deprecate a rule just because of its name?
One rule I would like rename is `noMultipleSpacesInRegularExpression`.

@ematipico @unvalley 
Do you prefer `noRedundnatElse` over `noUselessElse`? Some reasons?

**Comment by unvalley** [comment]
> Not sure if we should deprecate a rule just because of its name?

I guess we shouldn't. Editing biome.json by name change is kinda bothering.

> Do you prefer noRedundnatElse over noUselessElse? Some reasons?

As for me it's just preference. No strong reason.

> Also, Do you we should recommend the rule?

Nice! This rule helps readability and reducing waste code.

**Comment by ematipico** [comment]
I don't have strong opinions nor preferences. I like both. 

I definitely agree we should settle on one single naming style, and updating the internal convention. 

For renaming existing rules, that's breaking material, so we can worry about it later in the future.

**Comment by Conaclos** [comment]
> I definitely agree we should settle on one single naming style, and updating the internal convention, so new rules will follow it.

Ok, I will keep `Useless` because it is the most used form.

> For renaming existing rules, that's breaking material, so we can worry about it later in the future.

Yes. However, I think earlier the breaking change is, better is.

**Pull request merged**